gocheck2/checkers.go

```diff
@@ -138,6 +138,44 @@ func (h *hasKey) Info() *CheckerInfo {
 var HasKey = &hasKey{}
 
 // -----------------------------------------------------------------------
+// NoErr checker.
+
+type noErr struct{}
+
+// This exists to implement fmt.GoStringer and force the `%#v` format to show
+// the string unescaped, newlines and all.
+type rawString string
+
+func (r rawString) GoString() string {
+	return string(r)
+}
+
+func (c noErr) Check(params []interface{}, names []string) (bool, string) {
+	err, ok := params[0].(error)
```

Comment:
Yes, in that case `ok` will still be false. See http://play.golang.org/p/QocSsI0_pU .

The spec is supposed to describe this situation here: https://golang.org/ref/spec#Type_assertions and it says so too, though in the next paragraph it then seems to say the opposite. :-/

I'll add a test.

Thanks for the review!